src/plots/gl2d/scene2d.js

```diff
@@ -522,6 +522,21 @@ proto.draw = function() {
                             glplot.pixelRatio
                 ];
 
+                // this needs to happen before the next block that deletes traceCoord data
+                // also it's important to copy, otherwise data is lost by the time event data is read
+                this.graphDiv.emit('plotly_hover', {
+                    points: [{
+                        trace: nextSelection.trace,
+                        dataCoord: nextSelection.dataCoord.slice(),
+                        traceCoord: nextSelection.traceCoord.slice(),
+                        textLabel: nextSelection.textLabel,
+                        color: nextSelection.color,
+                        name: nextSelection.name,
+                        hoverinfo: nextSelection.hoverinfo,
+                        screenCoord: nextSelection.screenCoord.slice()
```

Comment:
First of all, thanks for pointing me to the Mapbox analogue!

Re your question, if you mean the availability of inputs to `Fx.hover` from within `scene2d.js`: 

-`gd` is available
- `subplot` just equals to a constant `xy` for `scene2d.js` I believe
- `evt` isn't available because the DOM mouse event capture is done deeper in the `gl-vis` stack, but with some work in it probably I can expose it, or I can perhaps reconstruct needed `evt` data as I already have `traceCoords`, `screenCoords`, mouse button info etc. in `scene2d.js`.

Hopefully I didn't misunderstand your question. What I don't quite understand yet is how it's possible to generate the event data that you ask for (i.e. identical structure and content with the SVG charts) without doing something similar to what you did for Mapbox,  because doing so seems to require the execution of most if not all logic in `fx.hover`.